implementations/rust/ockam/ockam_api/src/nodes/service/portals.rs

```diff
@@ -405,7 +427,34 @@ impl NodeManagerWorker {
             Ok(Response::not_found(req.id()).body(OutletStatus::new(
                 "".to_string(),
                 "".to_string(),
-                alias.clone(),
+                alias,
+                Some(format!("Outlet with alias {alias} not found").into()),
+            )))
+        }
+    }
+
+    pub(super) async fn show_outlet<'a>(
+        &mut self,
+        req: &Request<'_>,
+        alias: &'a str,
+    ) -> Result<ResponseBuilder<OutletStatus<'a>>> {
+        let node_manager = self.node_manager.write().await;
```

Comment:
You could use a read lock here: `self.node_manager.read().await;`

Reply:
Added in latest commit